state-chain/chains/src/btc/api.rs

```diff
@@ -146,6 +147,38 @@ impl<E: ReplayProtectionProvider<Bitcoin>> ExecutexSwapAndCall<Bitcoin> for Bitc
 	}
 }
 
+impl<E: ReplayProtectionProvider<Bitcoin>> RejectCall<Bitcoin> for BitcoinApi<E>
+where
+	E: ChainEnvironment<UtxoSelectionType, SelectedUtxosAndChangeAmount>
+		+ ChainEnvironment<(), AggKey>,
+{
+	type DepositDetails = <Bitcoin as Chain>::DepositDetails;
+	fn new_unsigned(deposit_details: Self::DepositDetails) -> Result<Self, RejectError> {
+		if let Some(utxo) = E::lookup(UtxoSelectionType::TakeUtxo(deposit_details.utxo_id)) {
+			if utxo.0.len() != 1 {
```

Comment:
We should not need to do any lookups in the environment.

The deposit_details (or whatever we pass in) needs to contain the utxo that we want to reject and the refund address (probably the fees too).

We want to construct a bitcoin transaction that sends the whole output back to source. So we will need the utxo and the refund address. We take the input utxo, and construct an output utxo with destination 'refund_address' and amount 'input - fees'.

Importantly, we don't want to receive any change from this operation because it's tainted, so we don't need to add a change utxo.